test/itkTorchImageTest.cxx

```diff
@@ -66,203 +133,160 @@ int itkTorchImageTest(int argc, char *argv[])
-
-    image->SetPixel( location0, thirdValue );
-    pixelValue = image->GetPixel( location0 );
-    itkAssertOrThrowMacro( pixelValue == thirdValue, StructName + "::SetPixel failed" );
+    const int SizePerDimension = 20;
+    const typename PixelType::ValueType firstValue[] = {1, 1, 1};
+    const typename PixelType::ValueType secondValue[] = {2, 2, 2};
+    const typename PixelType::ValueType thirdValue[] = {2, 3, 1};
+    itkTorchImageTestByTypeAndDimension< PixelType, ImageDimension >( SizePerDimension, StructName, firstValue, secondValue, thirdValue );
+  }
+  {
+    constexpr int VectorDimension = 2;
+    using PixelType = itk::Vector< short, VectorDimension >;
+    constexpr int ImageDimension = 2;
+    const std::string StructName = "TorchImage<Vector<short, 2>, 2>";
+    const int SizePerDimension = 250;
+    const typename PixelType::ValueType firstValue[VectorDimension] = {1, 2};
+    const typename PixelType::ValueType secondValue[VectorDimension] = {-100, 32000};
+    const typename PixelType::ValueType thirdValue[VectorDimension] = {100, -32000};
+    itkTorchImageTestByTypeAndDimension< PixelType, ImageDimension >( SizePerDimension, StructName, firstValue, secondValue, thirdValue );
+  }
 
-    ImageType::Pointer image2 = ImageType::New();
-    image2->SetRegions( size );
-    image2->Allocate();
-    image2->Graft( image );
+  {
+    constexpr int VectorDimension = 3;
+    using PixelType = itk::Vector< short, VectorDimension >;
+    constexpr int ImageDimension = 4;
+    const std::string StructName = "TorchImage<Vector<short, 3>, 4>";
+    const int SizePerDimension = 12;
+    const typename PixelType::ValueType firstValue[VectorDimension] = {1, 2, 3};
+    const typename PixelType::ValueType secondValue[VectorDimension] = {-310, 3100, -31000};
+    const typename PixelType::ValueType thirdValue[VectorDimension] = {310, -3100, 31000};
+    itkTorchImageTestByTypeAndDimension< PixelType, ImageDimension >( SizePerDimension, StructName, firstValue, secondValue, thirdValue );
+  }
+  {
+    constexpr int VectorDimension = 4;
+    using PixelType = itk::CovariantVector< short, VectorDimension >;
+    constexpr int ImageDimension = 5;
+    const std::string StructName = "TorchImage<CovariantVector<short, 4>, 5>";
+    const int SizePerDimension = 4;
+    const typename PixelType::ValueType firstValue[VectorDimension] = {1, 2, 3, 4};
+    const typename PixelType::ValueType secondValue[VectorDimension] = {-310, 3100, -31000, 31};
+    const typename PixelType::ValueType thirdValue[VectorDimension] = {310, -3100, 31000, 31};
+    itkTorchImageTestByTypeAndDimension< PixelType, ImageDimension >( SizePerDimension, StructName, firstValue, secondValue, thirdValue );
+  }
+  {
+    constexpr int VectorDimension1 = 2;
+    constexpr int VectorDimension2 = 3;
+    using PixelType = itk::Vector< itk::Vector< itk::RGBPixel< unsigned char >, VectorDimension1 >, VectorDimension2 >;
```

Comment:
Quite a convoluted test case! It should test quite a few less-exercised things.

Reply:
Yes, don't try this at home.  :-)